### ISSUE TITLE
Feat: Enforce null checks for datalist functions

### ISSUE THREAD

**Issue by n8-dev** [silverstripeltd/bespoke-standards#2]
Add a custom SS sniff for

Enforce null checks for datalist functions

This might need to be achieved with a static analyser https://silverstripe.atlassian.net/wiki/spaces/DEV/blog/2021/09/23/2414149848/Static+analyser+for+PHP+with+PHPStan

What I thought it would be is this

Disallow check for just a variable is true and ensure it is the expected data type
```
$page = SiteTree::get()->first();
if ($page) {

}
```
Disallow
```
$page = SiteTree::get()->first();
if ($page->exists()) {

}
```

Allow 

```
$page = SiteTree::get()->first();
if ($page instanceof SiteTree && $page->exists()) {

}
```

**Comment by edwilde** [comment]
Can you give an example of the before/after code?

**Comment by satrun77** [comment]
That was a suggestion from @blueo  correct me if I miss understood it.

**Comment by blueo** [comment]
yeah this was mainly intended to catch the return type of some data list functions eg 'first' 
![image](https://github.com/silverstripeltd/bespoke-standards/assets/948122/6ea0867f-957b-4d14-83fe-94f4e88a4d6c)
a basic null check would be ideal eg 

disallow
```PHP
$page = SiteTree::get()->first();
$title = $page->Title;
```

allow
```PHP
$page = SiteTree::get()->first();
if ($page === null) {
    return;
}

$title = $page->Title;
```

it prevents getting errors like "cannot access property of NULL "

but not sure how possible it is with phpcs

**Comment by satrun77** [comment]
That is what I thought to prevent the assumption of an object exists. It is not possible in phpcs but phpstan would do it out of box. :)

**Comment by GuySartorelli** [comment]
You could also use `$obj?->exists()` instead of `$obj && $obj->exists()` as it's a little more succinct. The null safety operator was added in PHP 8.0

**Comment by satrun77** [comment]
I would prefer to see code that does not rely on the Silverstripe framework and checks for the instance of the class instead of assuming it is the framework DataObject.
